substrate-archive/src/wasm_tracing.rs

```diff
@@ -197,6 +199,20 @@ impl TraceHandler {
 		self.span_events.lock().spans.push(span);
 		Ok(())
 	}
+
+	/// Start tracing with the predicate `fun`.
+	/// Consumes this TraceHandler.
+	pub fn scoped_trace<T>(self, fun: impl FnOnce() -> Result<T>) -> Result<(Vec<SpanMessage>, Vec<EventMessage>, T)> {
+		let span_events = self.span_events.clone();
+		let dispatch = Dispatch::new(self);
+		let res = dispatcher::with_default(&dispatch, fun)?;
```

Comment:
dq: what is `res` here?

Reply:
`dispatcher::with_default` accepts a closure which returns `T`, and in case that is useful further up the chain of function calls we catch it and return it here as `res`(short for result, but maybe a different name would be better)

Reply:
Ok, good to know. For the record I think its clear that `res` => result.